AIP-7385 rename KFP to AIP plugin

talebzeghmi opened the pull request:
TODO
- [x] friendly error message for use of `python flow.py kfp run` to use `aip run`

talebzeghmi committed:
AIP-7385 rename KFP to AIP plugin

talebzeghmi committed:
friendly error message for "kfp run"

cavandervoort commented on .github/workflows/test.yml:
```diff
@@ -3,12 +3,13 @@ on:
     - master
-    - feature/kfp
+    - feature/aip
+    - feature/aip
```
Is it supposed to say "- feature/aip" twice?

talebzeghmi replied:
fixed, thanks!

alexlatchford commented on metaflow/plugins/aip/kfp_cli.py:
```diff
@@ -0,0 +1,18 @@
+
+@kubeflow_pipelines.command(help="Deprecated, please use aip instead.")
+def run():
+    raise AIPException('Deprecated, please use "aip run" instead.')
```
Nice! Just to confirm is this the only `python flow.py kfp *` CLI action we support? I think so but just wanted to triple check :) 

Stretch goal: Maybe expand the error message to be something like `python myflow.py aip run` or to capture the flow name and output it exactly might be the icing on the cake so they can copy paste exactly what they had in old versus new? (likely w/o arguments but the top level might be nice)

talebzeghmi replied:
done, thanks!

alexlatchford commented on metaflow/plugins/aip/aip_metaflow_step.py:
```diff
@@ -335,7 +335,7 @@ def kfp_metaflow_step(
         or metaflow_configs_new["METAFLOW_USER"] is None
     ):
-        metaflow_configs_new["METAFLOW_USER"] = "kfp-user"
+        metaflow_configs_new["METAFLOW_USER"] = "aip-user"
```
Hmm this might be somewhat problematic for Flow Triggering Flow. I presume we'd need to support both `kfp-user` and `aip-user` in the RBAC bindings for a while. Should be easy enough but we'd need to make changes in `kubeflow-k8s-profiles` and in ZAP to cater to this change I'd imagine.

Likely just needs a Jira ticket and should be fairly quick to do and worthwhile to clean up this minor naming debt.

talebzeghmi replied:
see https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/kubeflow-k8s-profiles/-/merge_requests/902 !

alexlatchford commented on metaflow/plugins/aip/tests/flows/resources_flow.py:
```diff
@@ -84,7 +84,7 @@ def get_env_vars(env_resources: Dict[str, str]) -> List[V1EnvVar]:
 
 labels = {
-    "aip.zillowgroup.net/kfp-pod-default": "KF_POD_DEFAULT",
+    "aip.zillowgroup.net/aip-pod-default": "KF_POD_DEFAULT",
```
I think I asked this on another PR but do we still need this label? cc @aaron-arellano

aaron-arellano replied:
this can be removed. I am not sure why the kfp-pod-default was being maintained. We use an argo specific label to target workflow pods in the webhook.

alexlatchford commented on metaflow/plugins/aip/tests/run_integration_tests.py:
```diff
@@ -49,13 +49,13 @@
-    "kfp_flow.py",  # kfp_preceding_component feature has been deprecated.
+    "aip_flow.py",  # kfp_preceding_component feature has been deprecated.
     "flow_triggering_flow.py",  # TODO(talebz): will re-enable with Argo FTF
     # TODO(talebz) AIP-6717 re-enable for compilation changes or when cluster can handle
```
I see the lines below are commented out, is this work planned ahead of the release?

talebzeghmi replied:
tracking to re-disable nested foreach tests after kfp-argo merge with [AIP-7772](https://zillowgroup.atlassian.net/browse/AIP-7772)

alexlatchford reviewed (commented):
This is a cathartic PR 😄 Thanks for driving this migration, great to see us getting to this point!

As for the feedback, all largely minor stuff or clarifying comments but a couple of sticking ones would love to understand your position before I approve.

cloudw commented on .github/workflows/test.yml:
```diff
@@ -3,12 +3,13 @@ on:
     - master
-    - feature/kfp
+    - feature/aip
+    - feature/aip
```
We can probably keep feature/kfp branch around for a while, just in case we need to revisit them in unforeseeable situations.

talebzeghmi replied:
fixed, thanks!

cloudw commented on metaflow/plugins/aip/tests/test_argo_support.py:
```diff
@@ -8,7 +8,7 @@
 
 disabled_test_flows = [
-    "kfp_flow.py",  # kfp_preceding_component feature has been deprecated.
+    "aip_flow.py",  # kfp_preceding_component feature has been deprecated.
```
We should be ready to completely remove kfp_preceding_component now, right?

talebzeghmi replied:
I kept it because adding this feature was very complicated, and re-adding it would be too.
Instead we are:
* not exposing it to customers
* we'll remove by migrating to Metaflow Argo plugin, or by refactoring WFSDK AIP plugin to not use kfp compiler.  I added our next steps as a subject to discuss in our offsite.

talebzeghmi committed:
AIP-7393 Add Flow labels to Workflow for ArgoUI (#259)

* AIP-7393 Add Flow labels to Workflow for ArgoUI

* PR feedback

* AIP-7726 add default cards (#260)

* AIP-7726 add default cards

* param fix

* PR feedback
- comment archive none
- add card id to support current.card.append()
- add example of notebook card & multiple cards

* aip.zillowgroup.net/aip-pod-default -> aip.zillowgroup.net/aip-wfsdk-pod

* logging.zgtools.net/index -> logging.zgtools.net/topic

* fix card pathspec for nested foreach

* fix github test.yml branches

* resources_flow.py label fixes

talebzeghmi committed:
PR Feedback

talebzeghmi committed:
update test.yml
